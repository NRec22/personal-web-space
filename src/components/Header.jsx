import React from 'react';
import Navbar from 'react-bulma-components/lib/components/navbar';
import Container from 'react-bulma-components/lib/components/container';

const Header = () => (
  <Navbar>
    <Container>
      <Navbar.Brand>
        <Navbar.Item>...</Navbar.Item>
        <Navbar.Burger />
      </Navbar.Brand>
      <Navbar.Menu>
        <Navbar.Container>
          <Navbar.Item>Home</Navbar.Item>
          <Navbar.Item>About Me</Navbar.Item>
          <Navbar.Item>LinkedIn</Navbar.Item>
          <Navbar.Item>Github</Navbar.Item>
          <Navbar.Item>Contact Me</Navbar.Item>
        </Navbar.Container>
      </Navbar.Menu>
    </Container>
  </Navbar>
);

export default Header;