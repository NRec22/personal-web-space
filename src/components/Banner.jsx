import React from 'react';
import Hero from 'react-bulma-components/lib/components/hero';
import Container from 'react-bulma-components/lib/components/container';
import Heading from 'react-bulma-components/lib/components/heading';
import Header from './Header';

const Banner = () => (
  <Hero color="dark" size="medium" gradient>
    <Header />
    <Hero.Body>
      <Container>
        <Heading spaced size={1}>Neven Recchia</Heading>
        <Heading subtitle size={5}>Aspiring developer.</Heading>
      </Container>
    </Hero.Body>
  </Hero>
)

export default Banner;