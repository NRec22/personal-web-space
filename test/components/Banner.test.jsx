import { shallow, render } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Hero from 'react-bulma-components/lib/components/hero';
import Banner from '../../src/components/Banner';
import Header from '../../src/components/Header';

describe('Banner: ', () => {
  it('contains a hero body', () => {
    const wrapper = shallow(<Banner />);
    expect(wrapper.find(Hero)).toHaveLength(1);
  })

  it('should contain a header', () => {
    const wrapper = shallow(<Banner />);
    expect(wrapper.find(Header)).toHaveLength(1);
  })

  it('contains my name', () => {
    const wrapper = render(<Banner />);
    expect(wrapper.text()).toContain('Neven Recchia');
  })

  it('should render as expected', () => {
    const wrapper = shallow(<Banner />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})