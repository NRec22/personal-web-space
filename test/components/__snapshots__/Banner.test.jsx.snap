// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner:  should render as expected 1`] = `
<Hero
  className=""
  color="dark"
  gradient={true}
  renderAs="section"
  size="medium"
  style={Object {}}
>
  <Header />
  <Hero.Body
    className=""
    renderAs="div"
    style={Object {}}
  >
    <Container
      breakpoint={null}
      className=""
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Heading
        className=""
        heading={false}
        renderAs="h1"
        size={1}
        spaced={true}
        style={Object {}}
        subtitle={false}
      >
        Neven Recchia
      </Heading>
      <Heading
        className=""
        heading={false}
        renderAs="h1"
        size={5}
        spaced={false}
        style={Object {}}
        subtitle={true}
      >
        Developer at day. Thrillseeker at night.
      </Heading>
    </Container>
  </Hero.Body>
</Hero>
`;
